src/Time.ts

```diff
@@ -1,8 +1,24 @@
 import { Effect } from './Effect'
-import { handle, ok } from './Fx'
+import { Fx, handle, ok } from './Fx'
+import { ScheduledTask, TimeStep, TimeoutDisposable } from './internal/time'
 
-export class Now extends Effect('fx/Time')<void, number> { }
+export class Now extends Effect('fx/Time/Now')<void, number> { }
 
 export const now = new Now()
 
-export const builtinDate = handle(Now, () => ok(Date.now()))
+export class Monotonic extends Effect('fx/Time/Monotonic')<void, number> { }
+
+export const monotonic = new Monotonic()
+
+export class Schedule extends Effect('fx/Time/Schedule')<ScheduledTask, Disposable> { }
+
+export const schedule = (t: ScheduledTask) => new Schedule(t)
+
+export const defaultTime = <E, A>(f: Fx<E, A>): Fx<Exclude<E, Now | Monotonic | Schedule>, A> => f.pipe(
+  handle(Now, () => ok(Date.now())),
+  handle(Monotonic, () => ok(performance.now())),
+  handle(Schedule, ({ at, task }) => ok(new TimeoutDisposable(setTimeout(task, at))))
+) as Fx<Exclude<E, Now | Monotonic | Schedule>, A>
+
+export const stepTime = (nowOrigin = 0): TimeStep =>
```

Comment:
I'm not crazy about the name `stepTime`.  Suggestions welcome 😄

Reply:
Maybe we should just move it over to the `TimeStep` (or whatever name it) class as a static method like `make` or `create` ?